lib/history/sqlite/sqlite_test.go

```diff
@@ -198,50 +140,44 @@ func (s *SQLiteSuite) TestEviction(c *C) {
 func (s *SQLiteSuite) TestFilterEvents(c *C) {
 	var testCases = []struct {
 		comment  string
-		statuses []*pb.NodeStatus
+		events   []*pb.TimelineEvent
 		params   map[string]string
 		expected []*pb.TimelineEvent
 	}{
 		{
-			comment: "Expected one matching event.",
-			statuses: []*pb.NodeStatus{
-				&pb.NodeStatus{Name: "node-1", Status: pb.NodeStatus_Running},
-			},
-			params:   map[string]string{"type": nodeRecoveredType, "node": "node-1"},
-			expected: []*pb.TimelineEvent{history.NewNodeRecovered(s.clock.Now(), "node-1")},
+			comment:  "Expected one matching event.",
+			events:   []*pb.TimelineEvent{pb.NewNodeHealthy(s.clock.Now(), "node-1")},
+			params:   map[string]string{"type": string(history.NodeHealthy), "node": "node-1"},
+			expected: []*pb.TimelineEvent{pb.NewNodeHealthy(s.clock.Now(), "node-1")},
 		},
 		{
-			comment: "Expected no matching events.",
-			statuses: []*pb.NodeStatus{
-				&pb.NodeStatus{Name: "node-1", Status: pb.NodeStatus_Running},
-			},
-			params:   map[string]string{"type": nodeDegradedType, "node": "node-1"},
+			comment:  "Expected no matching events.",
+			events:   []*pb.TimelineEvent{pb.NewNodeHealthy(s.clock.Now(), "node-1")},
+			params:   map[string]string{"type": string(history.NodeDegraded), "node": "node-1"},
 			expected: nil,
 		},
 		{
 			comment: "Expected two matching events.",
-			statuses: []*pb.NodeStatus{
-				&pb.NodeStatus{Name: "node-1", Status: pb.NodeStatus_Running},
-				&pb.NodeStatus{Name: "node-1", Status: pb.NodeStatus_Degraded},
+			events: []*pb.TimelineEvent{
+				pb.NewNodeHealthy(s.clock.Now(), "node-1"),
+				pb.NewNodeDegraded(s.clock.Now().Add(time.Second), "node-1"),
 			},
 			params: map[string]string{"node": "node-1"},
 			expected: []*pb.TimelineEvent{
-				history.NewNodeRecovered(s.clock.Now(), "node-1"),
-				history.NewNodeDegraded(s.clock.Now().Add(time.Second), "node-1"),
+				pb.NewNodeHealthy(s.clock.Now(), "node-1"),
+				pb.NewNodeDegraded(s.clock.Now().Add(time.Second), "node-1"),
 			},
 		},
 	}
 
 	for _, testCase := range testCases {
+		testCase := testCase
```

Comment:
Why do you need this? As it looks, there're no goroutines which could use the reused value?